OptimizelySDK/OptimizelyUserContext.cs

```diff
@@ -184,10 +218,115 @@ public virtual void TrackEvent(string eventName)
         /// </summary>
         /// <param name="eventName">The event name.</param>
         /// <param name="eventTags">A map of event tag names to event tag values.</param>
-        public virtual void TrackEvent(string eventName, 
+        public virtual void TrackEvent(string eventName,
             EventTags eventTags)
         {
             Optimizely.Track(eventName, UserId, Attributes, eventTags);
         }
+
+        /// <summary>
+        /// Set a forced decision.
+        /// </summary>
+        /// <param name="context">The context object containing flag and rule key.</param>
+        /// <param name="decision">OptimizelyForcedDecision object containing variation key.</param>
+        /// <returns></returns>
+        public bool SetForcedDecision(OptimizelyDecisionContext context, OptimizelyForcedDecision decision)
+        {
+            lock (mutex)
+            {
+                ForcedDecisionsStore[context] = decision;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Gets a forced variation
+        /// </summary>
+        /// <param name="context">The context object containing flag and rule key.</param>
+        /// <returns>The variation key for a forced decision</returns>
+        public OptimizelyForcedDecision GetForcedDecision(OptimizelyDecisionContext context)
+        {
+            if (context == null || context.FlagKey == null)
+            {
+                Logger.Log(LogLevel.WARN, "flagKey cannot be null");
+                return null;
+            }
```

Comment:
Why do not have this checking in SetForcedDecision?

Reply:
We have this check in ForcedDecisionStore.